uitests/plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/requirement/CamelExampleRunner.java

```diff
@@ -0,0 +1,174 @@
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.eclipse.reddeer.common.condition.WaitCondition;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.jboss.tools.fuse.reddeer.utils.JarRunner;
+
+public class CamelExampleRunner extends JarRunner {
+
+	public static final Pattern IS_STARTED_PATTERN = Pattern.compile(".*Apache Camel .* started in .*");
+	public static final Pattern IS_SUSPENDED_PATTERN = Pattern.compile(".*Apache Camel .* is suspended in .*");
+	public static final Pattern IS_RESUMED_PATTERN = Pattern.compile(".*Apache Camel .* resumed in .*");
+
+	private CamelExampleOutput output;
+
+	public CamelExampleRunner(File jarFile) {
+		this(jarFile, false);
+	}
+
+	public CamelExampleRunner(File jarFile, File jolokiaJarFile) {
+		this(jarFile, false);
+	}
+
+	public CamelExampleRunner(File jarFile, boolean useJolokia) {
+		super(jarFile);
+		output = new CamelExampleOutput();
+		setOutput(output);
+
+		registerPattern(IS_STARTED_PATTERN);
+		registerPattern(IS_SUSPENDED_PATTERN);
+		registerPattern(IS_RESUMED_PATTERN);
+	}
+
+	@Override
+	public void run(String... args) {
+		super.run(args);
+		waitForOutputWitPattern(IS_STARTED_PATTERN);
+	}
+
+	@Override
+	public void stop() {
+		super.stop();
+	}
+
+	public void registerPattern(Pattern pattern) {
+		output.registerPattern(pattern);
+	}
+
+	public void waitForOutputWitPattern(Pattern pattern) {
+		new WaitUntil(new CamelExampleOutputContainsRegex(output, pattern), TimePeriod.LONG);
+	}
+
+	private class CamelExampleOutput implements Appendable {
+
+		public static final int LIMIT = 1000;
+
+		private List<String> lines;
+		private Set<Pattern> patternSet;
+		private Set<Pattern> patternFound;
+
+		public CamelExampleOutput() {
+			lines = new LinkedList<>();
+			patternSet = new HashSet<>();
+			patternFound = new HashSet<>();
+		}
+
+		public void registerPattern(Pattern pattern) {
+			patternSet.add(pattern);
+		}
+
+		public boolean isPatternFound(Pattern pattern) {
+			return patternFound.contains(pattern);
+		}
+
+		public List<String> getLines() {
+			return Collections.unmodifiableList(lines);
+		}
+
+		@Override
+		public String toString() {
+			StringBuffer buffer = new StringBuffer();
+			getLines().forEach(line -> buffer.append(line + System.getProperty("line.separator")));
+			return buffer.toString();
+		}
+
+		@Override
+		public Appendable append(CharSequence csq) throws IOException {
+			lines.add(csq.toString());
+			if (lines.size() > LIMIT) {
+				lines.remove(0);
+			}
+			for (Pattern pattern : patternSet) {
+				if (pattern.matcher(csq).matches()) {
+					patternFound.add(pattern);
+				}
+			}
+			return this;
+		}
+
+		@Override
+		public Appendable append(CharSequence csq, int start, int end) throws IOException {
+			return append(csq);
```

Comment:
It is a wrong implementation

> Appends a subsequence of the specified character sequence to this Appendable.
An invocation of this method of the form out.append(csq, start, end) when csq is not null, behaves in exactly the same way as the invocation `out.append(csq.subSequence(start, end))`

Reply:
The implementation was designed just for our concrete purpose, that's why it is private.

Reply:
but I can fix it, of course ;-)